scrnaseq.Rmd

```diff
@@ -25,56 +25,58 @@ library(reticulate) # required for 'leiden' clustering
 library(enrichR) # functional enrichment
 library(future) # multicore support for Seurat
 plan("multiprocess") # by default, this uses all available cores; use "workers=4" for example, to use 4 cores
+options(future.globals.maxSize = 800000000) # increase if the maximum allowed size in bytes of objects exported by future is exceeded
 
 # Other libraries we use
 # Knit: knitr
-# Data handling: dplyr, tidyr, purrr
+# Data handling: dplyr, tidyr, purrr, stringr, Matrix
 # Tables: kableExtra
 # Plots: ggsci, ggpubr
-# IO: openxlsx
+# IO: openxlsx, readr, R.utils
 # Annotation: biomaRt
 # DEG: mast
 # Functional enrichment: enrichR
-
-# Source plotting functions
-path_to_git = "~/scrnaseq"
-source(paste0(path_to_git, "/R/functions_io.R"))
-source(paste0(path_to_git, "/R/functions_plotting.R"))
-source(paste0(path_to_git, "/R/functions_analysis.R"))
-source(paste0(path_to_git, "/R/functions_util.R"))
+# Other: sessioninfo
 
 # Knitr default options
-knitr::opts_chunk$set(echo=TRUE, cache=FALSE, message=FALSE, warning=FALSE, fig.width=10)
+knitr::opts_chunk$set(echo=TRUE, cache=FALSE, message=FALSE, warning=FALSE, tidy=FALSE, fig.width=10)
 
 # Potentially needed for clustering, umap, other python packages
 use_python('/usr/bin/python')
 
-# biomaRt mirror
-mirror=NULL
+# biomaRt mirror: NULL means the first working mirror is used
+mirror = NULL
 ```
 
 # Dataset description
-* 10X published dataset, downloaded from [here](https://support.10xgenomics.com/single-cell-gene-expression/datasets/3.0.0/pbmc_1k_v3) 
-* Single Cell Gene Expression Dataset by Cell Ranger 3.0.0  
-* 1k Peripheral blood mononuclear cells (PBMCs) from a healthy donor (v3 chemistry)  
+* 10x and Smartseq2 datasets of PBMC cells
+* from "Systematic comparative analysis of single cell RNA-sequencing methods" `r knitcitations::citet("10.1186/s13059-019-1863-4")`
 
 ## Project-specific parameters
 This code chunk contains all parameters that are set specifically for your project. 
 ```{r project_parameters, class.source='fold-show'}
 param = list()
 
+# Path to git repository
+param$path_to_git = "."
```

Comment:
Can we move the source commands down to after this chunk? To not confuse setting parameters with sourcing scripts. Its minor, I can do it afterwards, too.

Reply:
done